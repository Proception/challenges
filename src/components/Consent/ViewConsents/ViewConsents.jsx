import React from 'react';
import Table from '../../Table/Table.jsx';

const ViewConsents = ({ consents }) => {
  const tableHeaders = ['Name', 'Email', 'Consent given for'];
  return (
    <Table tableHeaders={tableHeaders} tableRows={[...consents]} />
  );
};

export default ViewConsents;
