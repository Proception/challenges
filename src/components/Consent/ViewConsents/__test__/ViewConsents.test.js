import React from 'react';
import renderer from 'react-test-renderer';
import ViewConsents from '../ViewConsents.jsx';


describe('ViewConsents component', () => {
  it('should render ViewConsents correctly', () => {
    const consents = [
      {name: 'Benedict', email: 'omasan.esimaje@gmail.com', consents: ['Receive Newsletter','Be shown targeted ads']},
      {name: 'Omasan',email: 'benedict.esimaje@gmail.com',consents: ['Be shown targeted ads']},
      {name: 'Micheal',email: 'omasan@gmail.com', consents: ['Receive Newsletter', 'Be shown targeted ads']}
    ];
    const component = renderer.create(
      <ViewConsents consents={consents} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
