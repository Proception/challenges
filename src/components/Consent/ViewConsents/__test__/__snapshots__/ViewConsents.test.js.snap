// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewConsents component should render ViewConsents correctly 1`] = `
<div
  className="table-container"
>
  <div
    className="table-header"
  >
    <div
      className="table-column"
    >
      Name
    </div>
    <div
      className="table-column"
    >
      Email
    </div>
    <div
      className="table-column"
    >
      Consent given for
    </div>
  </div>
  <div
    className="table-body"
  >
    <div
      className="table-row"
    >
      <div
        className="table-column"
      >
        Benedict
      </div>
      <div
        className="table-column"
      >
        omasan.esimaje@gmail.com
      </div>
      <div
        className="table-column"
      >
        Receive Newsletter, Be shown targeted ads
      </div>
    </div>
    <div
      className="table-row"
    >
      <div
        className="table-column"
      >
        Omasan
      </div>
      <div
        className="table-column"
      >
        benedict.esimaje@gmail.com
      </div>
      <div
        className="table-column"
      >
        Be shown targeted ads
      </div>
    </div>
  </div>
  <div
    className="table-footer"
  >
    <div
      className="table-footer"
    >
      <div
        className="table-footer-column"
      >
        <a
          className="page-navigators"
          onClick={[Function]}
        >
          
        </a>
      </div>
      <div
        className="table-footer-column"
      >
        <a
          className="page-navigators page-number-spacing active-page"
          onClick={[Function]}
          value={
            Array [
              "1",
            ]
          }
        >
          1
        </a>
        <a
          className="page-navigators page-number-spacing "
          onClick={[Function]}
          value={
            Array [
              "2",
            ]
          }
        >
          2
        </a>
      </div>
      <div
        className="table-footer-column"
      >
        <a
          className="page-navigators"
          onClick={[Function]}
        >
          Next page  &gt;&gt;
        </a>
      </div>
    </div>
  </div>
</div>
`;
